OtripleS.Web.Api/Controllers/TeacherAttachmentsController.cs

```diff
@@ -0,0 +1,158 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using OtripleS.Web.Api.Models.TeacherAttachments;
+using OtripleS.Web.Api.Models.TeacherAttachments.Exceptions;
+using OtripleS.Web.Api.Services.TeacherAttachments;
+using RESTFulSense.Controllers;
+
+namespace OtripleS.Web.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TeacherAttachmentsController : RESTFulController
+    {
+        private readonly ITeacherAttachmentService teacherAttachmentService;
+
+        public TeacherAttachmentsController(ITeacherAttachmentService teacherAttachmentService) =>
+            this.teacherAttachmentService = teacherAttachmentService;
+
+        [HttpPost]
+        public async ValueTask<ActionResult<TeacherAttachment>> PostTeacherAttachmentAsync(
+            TeacherAttachment teacherAttachment)
+        {
+            try
+            {
+                TeacherAttachment persistedTeacherAttachment =
+                    await this.teacherAttachmentService.AddTeacherAttachmentAsync(teacherAttachment);
+
+                return Ok(persistedTeacherAttachment);
+            }
+            catch (TeacherAttachmentValidationException teacherAttachmentValidationException)
+                when (teacherAttachmentValidationException.InnerException is AlreadyExistsTeacherAttachmentException)
```

Comment:
We need to also handle where the `TeacherAttachment` has a reference to a non-existent entity.